### PR TITLE
🐛 fix(back): PLAS-046: Messaging Bugs

### PR DESCRIPTION
**Pull request by bvelitchkine**
Fixed a few messaging bugs due to useless code in the back + tried to limit errors in the service worker console, just a bit

### CODE REVIEW

File: apps/linkedin-to-notion/src/background/index.ts [hunk 1]
```diff
@@ -3,14 +3,16 @@ export {};
 const linkedInURLRegex = /linkedin.com\/in\/.+/;
 
 chrome.action.onClicked.addListener((tab) => {
-  if (tab.url.match(linkedInURLRegex)) {
+  if (tab.url?.match(linkedInURLRegex)) {
```
Comment (GauvainThery) [inline apps/linkedin-to-notion/src/background/index.ts:6]: Tab.url cannot be undefined so for me the `?` brings no value. But feel free to share the example of what you fixed doing so 😬

File: apps/linkedin-to-notion/src/background/index.ts [hunk 2]
```diff
@@ -3,14 +3,16 @@ export {};
...
     chrome.tabs.sendMessage(tab.id, 'toggleLinkedInNotionSidePanel');
   }
 });
 
-chrome.tabs.onUpdated.addListener((tabId, _, tab) => {
-  if (tab.url.match(linkedInURLRegex)) {
-    return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
+chrome.tabs.onUpdated.addListener((tabId, { status }, tab) => {
+  if (status === 'complete') {
+    if (tab.url?.match(linkedInURLRegex)) {
```
Comment (GauvainThery) [inline apps/linkedin-to-notion/src/background/index.ts:13]: Same here

File: apps/linkedin-to-notion/src/background/index.ts [hunk 3]
```diff
@@ -3,14 +3,16 @@ export {};
...
+  if (tab.url?.match(linkedInURLRegex)) {
     chrome.tabs.sendMessage(tab.id, 'toggleLinkedInNotionSidePanel');
   }
 });
 
-chrome.tabs.onUpdated.addListener((tabId, _, tab) => {
-  if (tab.url.match(linkedInURLRegex)) {
-    return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
+chrome.tabs.onUpdated.addListener((tabId, { status }, tab) => {
+  if (status === 'complete') {
```
Comment (GauvainThery) [inline apps/linkedin-to-notion/src/background/index.ts:12]: Good addition I think 🆗

**Review by GauvainThery — CHANGES_REQUESTED** [review]
I don't understand the purpose of the PR 🙈 

I think we have some errors in the service worker for those reasons:
- We have files in the contents folder that don't have a `PlasmoConfig` object so the extension spreads out of LinkedIn boundaries. Not urgent IMHO ➡️ can be fixed latter
- Sometime the update of the form is not done when going on a profile for the first time (or going on a profile then to another page and coming back on a profile) but this could go in a separated issue that would also allow to use pageHistory navigation to refresh the values in the form

TL;DR
Please give me more context as I don't really see an issue and show me in what sense your fix are fixing it 😬

File: apps/linkedin-to-notion/src/background/index.ts (hunk 1 above)
Comment (bvelitchkine) [reply apps/linkedin-to-notion/src/background/index.ts:6]: Yeah you're right. You already told me that but once again I got confused, my bad.

**Comment by bvelitchkine** [comment]
~Like I said in a comment above, I got confused. Just forget this PR!~

What should we do with this one? You seem to agree that the question mark has a purpose (cf. [your comment](https://github.com/Bastien-and-Gauvain/monorepo/pull/24#discussion_r1348782392) in this PR), when I seemed to have forgotten myself aha

**Comment by GauvainThery** [comment]
> ~Like I said in a comment above, I got confused. Just forget this PR!~
> 
> What should we do with this one? You seem to agree that the question mark has a purpose (cf. [your comment](https://github.com/Bastien-and-Gauvain/monorepo/pull/24#discussion_r1348782392) in this PR), when I seemed to have forgotten myself aha

Just close the PR, we'll fix that elsewhere 👌
